heron/scheduler-core/tests/java/com/twitter/heron/scheduler/SubmitterMainTest.java

```diff
@@ -124,21 +140,33 @@ public void testSubmitTopology() throws Exception {
     when(config.getStringValue(ConfigKeys.get(UPLOADER_CLASS))).thenReturn(UPLOADER_CLASS);
 
     // Failed to validate the submission
-    doReturn(false).when(submitterMain)
+    doThrow(new TopologySubmissionException(
+        String.format("Topology %s already exist", topology.getName()))).when(submitterMain)
         .validateSubmit(any(SchedulerStateManagerAdaptor.class), anyString());
-    assertFalse(submitterMain.submitTopology());
+
+    try {
+      submitterMain.submitTopology();
+      fail("submitTopology should throw exception");
+    } catch (TopologySubmissionException e) {
+      assertEquals(String.format("Topology %s already exist", topology.getName()), e.getMessage());
```

Comment:
Regarding "I am testing if the exception will be thrown out," the exception will always be throw because you added this to your unit test:
```
doThrow(new TopologySubmissionException(
        String.format("Topology %s already exist", topology.getName()))).when(submitterMain)
```
so you're not in fact testing that the logic within submitterMain will do this on it's own. You're testing logic you've inserted in the unit test.